src/ngraph/op/allreduce.cpp

```diff
@@ -27,13 +27,9 @@ op::AllReduce::AllReduce(const shared_ptr<Node>& arg)
 
 void op::AllReduce::validate_and_infer_types()
 {
-    if (validate_punt_if_dynamic())
-    {
-        return;
-    }
-
     NODE_VALIDATION_ASSERT(this,
-                           get_input_element_type(0) == element::f32 ||
+                           get_input_element_type(0).is_dynamic() ||
```

Comment:
Shouldn't this be `is_real()` ?

Reply:
The `is_dynamic()` is there to cover the case where the element type is not determined yet, in which case type checking "provisionally" passes. At runtime, `AllReduce` only allows `f32` and `f64`.